Source/player.cpp

```diff
@@ -2145,8 +2145,11 @@ void LoadPlrGFX(Player &player, player_graphic graphic)
 		szCel = "qm";
 		break;
 	case player_graphic::Death:
-		if (animWeaponId != PlayerWeaponGraphic::Unarmed)
-			return;
+		if (animWeaponId != PlayerWeaponGraphic::Unarmed) {
+			assert("LoadPlrGFX: requested Death animation with non-unarmed weapon");

```

Comment:
This is not how an assert is supposed to work. Instead, you would do:
`assert(animWeaponId == PlayerWeaponGraphic::Unarmed);`

And if the condition is false, the game will crash with an error message. If you pass in a string, the condition will always be true because a non-empty string evaluates to true in Boolean expressions. Effectively, this is a no-op.

However, I don't think we should be using an assert here, based on what I now know about the network messages that can lead to this situation. Perhaps it should be a comment.

```suggestion
			// Only one Death animation exists, for unarmed characters
```